### ISSUE TITLE
Show clearer feedback (snackbar, etc) when applicant hits Next but has unfilled questions

### ISSUE THREAD

**Issue by neel-a-mehta** [civiform/civiform#1474]
When you're filling out an application and hit Next without filling in all the required fields, the operation just fails silently. Some red error text appears near unfilled questions, but those may be all the way up the page. There's little feedback given around the Next button itself. This is confusing for screen readers, since they can't see the changes halfway up the page, and even for users without screen readers, since there's no immediate way of knowing why the Next button is failing.

Instead, we could:

- Gray out the Next button until all required fields are filled
- Reload the page and show a prominent error message (such as in snackbar) 
- Go back to the part of the page that had an error (scrolling?) and focus on the error messages so that the screen reader will read it.

This problem is mostly apparent with Name and Address questions, which have small red validation text way up the page.

## Tester's original report 

> Actual results:
At Step 6, observe error messages / text are displayed in red for input text fields like 'address line 1' , 'city', 'state', 'zip code', 'first name' & 'last name' but screen reader does not announce the error messages.
> Expected results: Error's , warnings visually being displayed should also be notified for screen reader users appropriately.

**Comment by neel-a-mehta** [comment]
Something similar happens with Enumerator questions: if you don't fill in any enumerated values but then hit Next, it fails silently. Some error text pops up near the enumerator input fields, but nothing around the Next button.

![image](https://user-images.githubusercontent.com/77128518/123302742-dd3dcb80-d4d1-11eb-835a-ca1de1a848c0.png)

I think the smoothest solution in this case is to disable the Next button until someone properly fills in at least one enumerated value.

**Comment by neel-a-mehta** [comment]
Now the same thing happens with the File Upload button. If I hit Upload without uploading a file, the form does not submit, but the red error message is not next to the Upload button, making it potentially confusing for screen readers.

![image](https://user-images.githubusercontent.com/77128518/123314582-9d7de080-d4df-11eb-889f-d74d1f7a6d53.png)

Like before, I think the move here is to disable the Upload button and gray it out until a file is uploaded, so the user knows (even if they're using a screen reader) that they can't go forward without doing something (or skipping).

**Comment by jeanniefu** [comment]
Closing in favor of https://github.com/civiform/civiform/issues/3563, which suggests a fix recommended by the a11y team for this described issue.
